### PR TITLE
RHOAIENG-12195: Update TrustyAI dependencies

### PR DESCRIPTION
**Pull request by ruivieira**
- See [RHOAIENG-12195](https://issues.redhat.com/browse/RHOAIENG-12195)
- Update TrustyAI to 0.6.1 (adds support for TMaRCo local models, see [RHOAIENG-12554](https://issues.redhat.com/browse/RHOAIENG-12554))
- Add PyTorch, transformers, datasets and accelerate as TrustyAI notebook image dependencies

## Description

In order to add support for disconnected use and for local models in TrustyAI's TMaRCo/Detoxification, this PR changes the following:

- Update TrustyAI version to 0.6.1
- Add the PyTorch, transformers, datasets and accelerate dependencies. These were previously optional.

## How Has This Been Tested?

Updated example on using local models with TMaRCo to be added with https://github.com/trustyai-explainability/trustyai-explainability-python-examples/pull/43

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter/trustyai/ubi9-python-3.11/Pipfile [hunk 1]
```diff
@@ -3,11 +3,21 @@ url = "https://pypi.org/simple"
...
+[[source]]
+url = "https://download.pytorch.org/whl/cu121"
+verify_ssl = true
+name = "pytorch"
+
 [dev-packages]
 
 [packages]
+# PyTorch packages
+torch = {version = "~=2.2.1", index = "pytorch"}
```
Comment (jstourac) [inline jupyter/trustyai/ubi9-python-3.11/Pipfile:15]: Note: we should check how much the new images are bigger and whether we're okay with that.

Comment (ruivieira) [reply jupyter/trustyai/ubi9-python-3.11/Pipfile:15]: @jstourac According to a cursory look this increase the image size (on the registry) from ~1GiB to ~5Gib.

**Review by atheo89 — APPROVED** [review]
Thanks @ruivieira for your PR, it /lgtm :slightly_smiling_face: 
I was able to spin up the notebook via `podman run -it -p 8888:8888 quay.io/opendatahub/workbench-images:jupyter-trustyai-ubi9-python-3.11-pr-697` without any complains. TBH I didn't test it along with detoxification module.

**Pull request merged**